### ISSUE TITLE
Large performance gap between single gpu and 8 gpus training

### ISSUE THREAD

**Issue by yukang2017** [cheng052/BRNet#1]
Hi,

Thanks for your great work. I found there is a large performance gap between training on single gpu and 8 gpus on the sunrgbd dataset.

When I train the code on single gpu as the instruction in README.md, I got 
AP_0.25, AR_0.25, AP_0.50, AR_0.50
0.5941, 0.8539, 0.4211, 0.5965

However, when I train the code on 8 gpus with  tools/dist_train.sh, I got
AP_0.25, AR_0.25, AP_0.50, AR_0.50
0.5794, 0.8587, 0.3958, 0.5704

There exists a sudden performance drop when training on 8 gpus.

(1) Could you provide some instructions on how to get the normal results on 8 gpus? For example, how to set lr, batch size, syncbn?

(2) Even on single gpu, there is also one point performance gap from the reported 43.7 on mAP50. Is it just from the instability? Need I train multiple times to get better results?

Thanks in advance.

Regards,
Chen Yukang

**Comment by cheng052** [comment]
Hi,

Thanks for your interest in our work.

(1) Due to resource limitations, I don't try to train the code on 8 GPUs. And for the multi-GPU training, I think the following may work:
- lr setting: use a larger learning rate(#GPU * original lr), i.e. 0.008 for SUN RGB-D trained with 8 GPUs. And I think you may tune the lr using different values to get the best result.
- batch size: keep 8 samples_per_gpu, and use a total batch size of 64 for 8 GPUs.
- syncbn: I have little experience in multi-GPU training for BRNet, so I can't provide suggestions on syncbn, sorry for this.

(2) The performance gap on single GPU training: The detection results on ScanNet and SUN RGB-D are unstable, and sometimes fluctuate within 1~2 mAP points across different runs using the same setting. It's suggested to train multiple times to get better results.

Hope the suggestions above can answer your question.

Regards,
Bowen Cheng

**Comment by yukang2017** [comment]
Thanks for your suggestions.
